tests/functional/event_handler/test_api_middlewares.py

```diff
@@ -397,6 +397,46 @@ def dummy_route():
     assert result["statusCode"] == 200
 
 
+@pytest.mark.parametrize(
+    "app, event",
+    [
+        (ApiGatewayResolver(proxy_type=ProxyEventType.APIGatewayProxyEvent), API_REST_EVENT),
+        (APIGatewayRestResolver(), API_REST_EVENT),
+        (APIGatewayHttpResolver(), API_RESTV2_EVENT),
+    ],
+)
+def test_api_gateway_middleware_with_include_router_prefix(app: EventHandlerInstance, event):
+    # GIVEN a router instance
+    router = Router()
+
+    def app_middleware(app: EventHandlerInstance, next_middleware: NextMiddleware):
+        # inject a variable into resolver context
+        app.append_context(injected="injected_value")
+        response = next_middleware(app)
+
+        return response
+
+    # WHEN we register a route with a middleware
+    @router.get("/path", middlewares=[app_middleware])
+    def dummy_route():
+        # if the middleware is executed, the context variable will be available
+        # Otherwise, an exception will be raised
+        injected_context = app.context["injected"]
+
+        assert injected_context == "injected_value"
```

Comment:
```suggestion
        assert app.context["injected"] == "injected_value"
```

suggestion: this will now fail with KeyError Exception instead AssertionError, making it more explicit the behaviour we're testing and easier to identify why that's the case (back to the comment being THEN too).

Mental model is that the `app.context[...]` is the key we're supposedly injecting. If the key isn't injected, then the exception alone will tell us the problem.